### PR TITLE
added add & remove

### PR DESCRIPTION
**Pull request by AndresEspi**
### **Project:** To Do list - Add & Remove

Learning Objectives:

Use webpack to bundle JavaScript.
Apply proper ES6 syntax.
Utilize ES6 modules to write modular JavaScript.
Estimated Time: 4 hours

Description:
In this project, you will be implementing the CRUD (Create, Read, Update, Delete) methods to make the elements of the user interface fully functional. The application will allow users to add and remove tasks from a To-Do list. Make sure to follow all the requirements before you start building your project.

General Requirements:

Ensure there are no linter errors.
Follow the correct GitHub Flow.
Document your work in a professional manner.
HTML/CSS Requirements:

Follow the best practices for HTML & CSS.
JavaScript Requirements:

Follow the best practices for JavaScript.
Project Requirements:

Remove all hardcoded items from the tasks array.
Create a new JavaScript file for the new functionality.
Implement a function for adding a new task (add a new element to the array).
Implement a function for deleting a task (remove an element from the array).
Implement a function for editing task descriptions.
By default, new tasks should have the property "completed" set to false and the property "index" set to the value of the new array length (i.e., if you're adding a 5th task to the list, the index of that task should equal 5).
Deleting a task should update all remaining items' indexes, so they represent the current list order and are unique (i.e., if you're deleting the first task index 1 from the list, the index of the next task(2) should be set to 1).
All changes to the To-Do list should be saved in local storage.
Big Picture Reminder:
The overall goal of this project is to implement a functional To-Do list application, utilizing proper ES6 syntax and webpack for bundling. You'll be creating functions to add, remove, and edit tasks while ensuring the list state is preserved in local storage.

Work and Submission Mode:
This activity should be completed individually, and you should submit it accordingly.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -39,16 +20,138 @@ const renderList = (list) => {
...
+    task.index = index + 1;
+  });
+  saveToLocalStorage(todoList);
+  renderList(todoList);
+};
+
+const clearButton = document.querySelector('.clear-btn');
+clearButton.addEventListener('click', () => {
+  clearCompletedTasks();
+});
```
Comment (AmaduKamara) [inline src/index.js:84]: - [ ] Let's try to work on project requirements and build our app feature by feature as we're required to. The Clear All Completed feature is not a requirement for this milestone. Kindly remove this functionality as it's meant for a future milestone you'll be working on in the coming tasks. 👍🏽

**Review by AmaduKamara — CHANGES_REQUESTED** [review]
# Status: Required Changes ♻️ ♻️ ♻️

Hi @AndresEspi,

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✅ The Linter checks passed successfully 👍🏽
✅ Removed all hardcoded todos 👍🏽

## Required Changes ♻️
_Check the comments under the review._

- [ ] After I installed the packages, your app is failing to load properly and I get the errors in the screenshots below. Kindly work on it so the app loads correctly and all requirements are met.

Screenshot 2023-07-22 at 3 59 20 PM

Screenshot 2023-07-22 at 3 59 43 PM

- [ ] In your README file, the Live Demo link is linked to your Awesome Books project. Kindly remove that as this project is the Todo List app. You can place N/A at the moment until you complete this one and host it online, then you can add the link there. 👍🏽

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Review by IngDavidTM — CHANGES_REQUESTED** [review]
## Hi @AndresEspi 🙋‍♂️, 

Good job so far!
You have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!



## Top highlight 🎯
- The Pull Request follows the GitHub Flow ✅ 
- The PR title and description are professional and descriptive ✅ 
- Linters are working properly ✅ 

## Suggested changes

- [ ] While you did a fantastic job with your project, as mentioned by the previous reviewer after I installed the packages, your app is failing to load properly and I get the errors. Kindly work on it so the app loads correctly and all requirements are met :+1:.

Screenshot from 2023-07-22 12-02-21

- [x] _Check the comments under the review._ ⬇

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: webpack.config.js [hunk 2]
```diff
@@ -23,6 +23,16 @@ module.exports = {
...
       },
+      {
+        test: /\.js$/,
+        exclude: /node_modules/,
+        use: {
+          loader: 'babel-loader',
+          options: {
+            presets: ['@babel/preset-env'],
+          },
+        },
```
Comment (V-Blaze) [inline webpack.config.js:34]: - [ ] You have done great so far but there are still some issues you need to resolve. :+1: :+1: As pointed out by the previous reviewer, when I ran `npm run start`, I got errors in the browser and terminal as shown in the screenshot below, I'll suggest you go back to the lesson on setting up Webpack and try using it to troubleshoot the error. :wave: :+1: :smile: 

- - Screenshot:
 
Screenshot from 2023-07-22 19-03-32

File: src/index.js [hunk 3]
```diff
@@ -39,16 +20,124 @@ const renderList = (list) => {
...
+
+const editTaskDescription = (taskId, newDescription) => {
+  todoList.forEach((task) => {
+    if (task.id === taskId) {
+      task.description = newDescription;
+    }
+  });
+  saveToLocalStorage(todoList);
+};
+
```
Comment (V-Blaze) [inline src/index.js:71]: > Create a new JavaScript file for the new functionality.

- [ ] You have done great writing out the code for adding, removing and editing to-do items, but looking at the project requirement, you were meant to implement this in a new `.js` file. I'll suggest you create a folder with the name `module` then inside the folder create a new javascript file with a name of choice and then implement the functionality there. :+1: :wave: :100:

**Review by V-Blaze — CHANGES_REQUESTED** [review]
## Hi @AndresEspi , 

# Almost there 🟡 

1094a3e620ebea4cd6cf2d0aceaa8f52


Great work so far, You have come a long way, keep moving forward👏 👍  
You've done well implementing some changes from the previous reviewers, but there are still some issues I think you need to improve on. 😄 👋 


 ## Highlights
-  LInters setup correctly✔
- Correct GitHub flow💯 🥇 ✔

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/index.js (hunk 3 above)
Comment (AndresEspi) [reply src/index.js:71]: is this a mandatory change????=?

**Review by laiifuu — COMMENTED** [review]
## Hi @AndresEspi :wave: 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project where linters check aren't passing :x: 

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Review by skyv26 — APPROVED** [review]
## Hi @AndresEspi, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:  
Congratulations! 🎉 

## STATUS: APPROVED 🟢 




## Highlights
- Good **PR** title ✅ 
- Descriptive *PR* description ✅ 
- Implement the features ✅ 
- Professional `README` template ✅ 
- Linters are green ✅ 
- Correct `GitHub` workflow ✅ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

## ⚠️  WARNING ⚠️ 
**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

## Have any doubt ❓ 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @skyv26 in your question so I can receive the notification. You can also follow me on the below platforms

<table>
<tr>
<td>

<a href="https://github.com/skyv26">
   @skyv26
</a>

</td>
<td>

<a href="https://www.linkedin.com/in/skyv2022/">
   linkedin-skyv2022
</a>

</td>
<td>

<a href="https://twitter.com/vrma_aakash">
   @vrma_aakash
</a>

</td>
<td>

<a href="https://www.instagram.com/skybrel/">
   @skybrel
</a>

</td>
<td>

<a href="https://www.frontendmentor.io/profile/skyv26">
   @skybrel
</a>

</td>
</tr>
</table>

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**